packages/protocol/contracts/bridge/libs/LibBridgeProcess.sol

```diff
@@ -41,18 +44,19 @@ library LibBridgeProcess {
     /// @param state The state of the bridge.
     /// @param resolver The address resolver.
     /// @param message The message to be processed.
-    /// @param proof The proof of the message hash from the source chain.
+    /// @param proofs The list of merkle inclusion proofs.
     /// @param checkProof A boolean flag indicating whether to verify the signal
     /// receipt proof.
     function processMessage(
         LibBridgeData.State storage state,
         AddressResolver resolver,
         IBridge.Message calldata message,
-        bytes calldata proof,
+        bytes[] calldata proofs,
```

Comment:
Same mechanism shall be applied to `recallMessage` ? It would be a bit more complex as that one requires full merkle proof from the Birdge 🤔

Reply:
But it will be a bit more difficult, because we do not use the `SignalService` for `recallMessage` but full merkle proofs from the destination `Bridge`. But how can we do this '2 step' proving without it ? 🤔 
Maybe relaying over the `BrideRoot` (or the stateRoot of the L2) in LibVerifying.sol just like we relay the `signalRoot`:
```
                    ISignalService(resolver.resolve("signal_service", false))
                        .sendSignal(signalRoot);
```

Then we need 2 proofs: 1 signalProof, and 1 full merkle ? 🤔

Reply:
You are right, recallMessage shall also be updated accordingly. Let me do it later today.